local_planner/CMakeLists.txt

```diff
@@ -125,17 +139,32 @@ include_directories(
 # add_library(avoidance
 #   src/${PROJECT_NAME}/avoidance.cpp
 # )
-add_library(local_planner
-  src/nodes/local_planner.cpp
-  src/nodes/waypoint_generator.cpp
-  src/nodes/histogram.cpp
-  src/nodes/tree_node.cpp
-  src/nodes/box.cpp
-  src/nodes/star_planner.cpp
-  src/nodes/planner_functions.cpp
-  src/nodes/common.cpp
-  src/nodes/rviz_world_loader.cpp
-)
+
+if(DISABLE_SIMULATION)
```

Comment:
Rather than repeating this setup in 2 separate places, add this list of files to a variable, and only if required also append the rviz world loader. Then do the add_library outside of the if.